odl/trafos/backends/pywt_bindings.py

```diff
@@ -0,0 +1,523 @@
+        Specification of the wavelet to be used in the transform.
+        If a string is given, it is converted to a `pywt.Wavelet`.
+    pad_mode: string
+        ODL style boundary condition.
+    shape_list : `sequence`
+        Sequence containing the shapes of the wavelet (approximation
+        and detail) coefficients at different scaling levels.
+
+    Returns
+    -------
+    out : `numpy.ndarray`
+        Flattened and concatenated coefficient array.
+        The length of the array depends on the shape of the input image to
+        be transformed and on the chosen wavelet basis.
+    """
+    coeff = np.asarray(coeff)
+    wbasis = pywt_wbasis(wbasis)
+    coeff_list = array_to_pywt_coeff(coeff, shape_list)
+    pywt_mode = PAD_MODES_ODL2PYWT[pad_mode]
+    ndim = len(shape_list[0])
+    nscales = len(shape_list) - 1
+
+    if ndim == 1:
+        return pywt.waverec(coeff_list, wbasis, pywt_mode)
+    elif ndim == 2:
+        return pywt.waverec2(coeff_list, wbasis, pywt_mode)
+    elif ndim == 3:
+        return _waverec3(coeff_list, wbasis, pywt_mode, nscales)
+    else:
+        raise NotImplementedError('no transform available for {} dimensions'
+                                  ''.format(ndim))
+
+
+def _wavedec3(arr, wbasis, pywt_mode, nscales):
+    """Discrete 3D multiresolution wavelet decomposition.
+
+    Helper function for the 3D wavelet decomposition that constructs
+    the same type of ordered coefficient list as the 1D and 2D variants
+    of `pywt.wavedecn`. This is necessary since the 3D variant returns
+    a dictionary instead of an ordered list.
+
+    Parameters
+    ----------
+    arr : `array-like`
+        Input array to the wavelet decomposition.
+    wbasis :  string or `pywt.Wavelet`
+        Specification of the wavelet to be used in the transform.
+        If a string is given, it is converted to a `pywt.Wavelet`.
+    pywt_mode: string
+        `pywt` style boundary condition.
+    nscales : int
+        Number of scaling levels to be used in the transform.
+        The maximum number of usable scales can be determined
+        by `pywt.dwt_max_level`.
+
+    Returns
+    -------
+    coeff_list : ordered list
+        List of coefficients organized in the following way:
+
+            ```[aaaN, (aadN, adaN, addN, daaN, dadN, ddaN, dddN), ...,
+            (aad1, ada1, add1, daa1, dad1, dda1, ddd1)]```
+
+        The abbreviations are "a" for "approximation" and "d" for "detail".
+        The letter position encodes the coefficient type in the respective
+        axis. ``N`` stands for the number of scaling levels.
+    """
+    wbasis = pywt_wbasis(wbasis)
+    coeff_list = []
+
+    wcoeffs = pywt.dwtn(arr, wbasis, pywt_mode)
+    aaa = wcoeffs['aaa']
+    aad = wcoeffs['aad']
+    ada = wcoeffs['ada']
+    add = wcoeffs['add']
+    daa = wcoeffs['daa']
+    dad = wcoeffs['dad']
+    dda = wcoeffs['dda']
+    ddd = wcoeffs['ddd']
+
+    details = (aad, ada, add, daa, dad, dda, ddd)
```

Comment:
Yes, because this is the way they are generated [in `PyWavelets`](https://github.com/PyWavelets/pywt/blob/5fc0f769dfdd6d21c2d0b2fa4fa8cdb7e887bb9f/pywt/_multidim.py#L267).